zerver/lib/notification_data.py

```diff
@@ -352,4 +358,10 @@ def get_mentioned_user_group_name(
             smallest_user_group_size = current_user_group_size
             smallest_user_group_name = current_user_group.name
```

Comment:
Reading the code above, I think `get_user_group_direct_member_ids` might be the wrong check -- we should be checking the recursive membership of the group, not just the number of direct members. (Not important yet as we're not using nested groups in production yet, but we will be fairly soon).

Reply:
Hmm, right. `get_user_group_member_ids` is the correct function to use. Made the change as a prep commit. 

Tests included to verify the changes.